### DISCUSSION TITLE
Getting blank array for computing fractions of greenspace

### DISCUSSION THREAD

**Discussion by advythr** [earthlab-education/earth-analytics-02-python-2024#7] [Q&A01 Urban Greenspace | Cloud Data] [answered]
I'm running this script in Jupyter Notebook in VS Code on a 2021 MacBook Pro using the EA python env.

I have the following code, which is very similar to that in the tutorial video:

```
# Workflow for one tract

ndvi_stats_list = []

for tract, naip_item_df in item_url_df.groupby('tract'):
    tract_ndvi_das = []
    print(tract)
    for i, naip_url_s, in naip_item_df.iterrows():
        print(naip_url_s.url)

        # Open NAIP data array
        full_naip_vda = rxr.open_rasterio(naip_url_s.url, masked=True).squeeze()

        # Get census tract boundary
        boundary_gdf = tracts_clipped.to_crs(full_naip_vda.rio.crs)[tracts_clipped.TRACTCE==tract]
        
        # Clip NAIP data to boundary
        crop_naip_vda = full_naip_vda.rio.clip_box(
            *boundary_gdf.total_bounds
        )

        naip_vda = crop_naip_vda.rio.clip(boundary_gdf.geometry)

        # Compute NDVI
        # Band 4: NIR, Band 1: Red
        tract_ndvi_das.append(
            (naip_vda.sel(band=4) - naip_vda.sel(band=1))
            / (naip_vda.sel(band=4) + naip_vda.sel(band=1))
        )

    # Merge rasters if there are multiple images
    if len(tract_ndvi_das)>1:
        tract_ndvi_da = rxr.merge_arrays(tract_ndvi_das)
        print("Merged images")

    else:
        print("Only one image for tract")
        tract_ndvi_da = tract_ndvi_das[0]

    # Compute fraction of greenspace (NDVI>NDVI threshold)
    # print(tract_ndvi_da.notnull().sum())
    total_pixels = tract_ndvi_da.notnull().sum()

total_pixels
```

I get a blank array as output:

```
xarray.DataArray
array(281658)
Coordinates:
spatial_ref
()
int64
0
Indexes: (0)
Attributes: (0)
```

If I try to plot..
```
total_pixels.plot()
```

I get
```
TypeError: No numeric data to plot.
```

However if I plot tract_nvi_da itself,

<img width="792" alt="Screen Shot 2024-01-29 at 5 50 19 PM" src="https://github.com/earthlab-education/earth-analytics-02-python-2024/assets/83184164/6f993c98-08df-45ce-8d83-65d45c46deb9">

There are clearly non-null values.

If I only do the notnull() operation,
```
total_pixels = tract_ndvi_da.notnull()
```

I do see there are non-null values:
<img width="1018" alt="Screen Shot 2024-01-29 at 5 51 54 PM" src="https://github.com/earthlab-education/earth-analytics-02-python-2024/assets/83184164/1ba42929-158c-45d5-8362-e96395c2fce8">

So I'm thinking this is an issue with .sum()?

Anyone know what's happening here and how to fix it?

Thanks!

**Comment by advythr** [comment]
@eculler

**Comment by edchanonline** [comment, marked as answer]
It looks like you had valid values? Try the following!
![Screenshot 2024-01-29 at 6 51 46 PM](https://github.com/earthlab-education/earth-analytics-02-python-2024/assets/4795392/e95bd7b3-35b9-41cc-a4d7-f56b0e8654ef)

**Reply by advythr** [reply]
Thanks for the clarification and quick response, Ed!

You're correct, I had valid values but I was getting confused since the result is an array with only one value.
